### PR TITLE
feat: Add opt in webview that is shown prior to enabling the coding assistant

### PR DESCRIPTION
**Pull request by brendanator**
Review along with sourcery-ai/core#2918

### CODE REVIEW

File: src/chat.ts [hunk 1]
```diff
@@ -284,36 +284,36 @@ export class ChatProvider implements vscode.WebviewViewProvider {
...
-				
-				<link href="${styleResetUri}" rel="stylesheet">
-				<link href="${styleVSCodeUri}" rel="stylesheet">
-				<link href="${styleMainUri}" rel="stylesheet">
-				<link href="${animationsUri}" rel="stylesheet">
-				<link href="${hljsUri}" rel="stylesheet">
-
-			</head>
-			<body class="sidebar__chat-assistant-body">
+      <html lang="en" style="height: 100%">
```
Comment (brendanator) [inline src/chat.ts:287]: This is the first of several formats that just convert tabs to spaces

File: src/opt-in.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      width: 100%;
+      margin-top: 15px;
+      padding: 10px;
+      background-color: #007bff;
+      color: #fff;
+      border: none;
+      border-radius: 5px;
+      cursor: pointer;
+    }
+  </style>
```
Comment (bm424) [inline src/opt-in.ts:61]: **suggestion:** consider extracting this to a separate CSS file to match our other webviews.

File: src/opt-in.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
+"use strict";
+
+import { commands, WebviewView, WebviewViewProvider } from "vscode";
+
+type OptInMessage = {
+  command: "opt-in";
+};
```
Comment (bm424) [inline src/opt-in.ts:7]: **suggestion:** find a way to remove this interface and its construction for now - it's not necessary (the `postMessage` can just be called with a string or `null` for instance)

File: package.json [hunk 4]
```diff
@@ -51,37 +51,45 @@
...
+					"when": "sourcery.features.coding_assistant && !sourcery.config.coding_assistant.enabled"
+				},
 				{
 					"id": "sourcery.chat",
 					"name": "Coding Assistant",
 					"type": "webview",
 					"icon": "sourcery-icon.png",
 					"contextualTitle": "Coding Assistant",
-					"when": "sourcery.features.coding_assistant"
+					"when": "sourcery.features.coding_assistant && sourcery.config.coding_assistant.enabled === true"
```
Comment (bm424) [inline package.json:68]: **question:** is the `=== true` necessary here? Why?

File: package.json [hunk 5]
```diff
@@ -51,37 +51,45 @@
...
 				},
 				{
 					"id": "sourcery.troubleshooting",
 					"name": "Troubleshooting",
 					"type": "webview",
 					"icon": "sourcery-icon.png",
 					"contextualTitle": "Troubleshooting",
-					"when": "sourcery.features.coding_assistant && sourcery.features.coding_assistant_troubleshooting && sourcery.features.semantic_search"
+					"when": "sourcery.features.coding_assistant && sourcery.config.coding_assistant.enabled === true && sourcery.features.coding_assistant_troubleshooting && sourcery.features.semantic_search"
```
Comment (bm424) [inline package.json:76]: **issue:** I think this is a good example of one of the reasons I'm concerned with the approach taken in the core PR - despite the fact that the features in `core` now take configuration, it still has to be explicitly handled here - yuck.

File: src/opt-in.ts [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    </form>
+  </div>
+</body>
+<script>
+  function submitForm(event) {
+    event.preventDefault();
+    const vscode = acquireVsCodeApi();
+    vscode.postMessage({ command: 'opt-in' });
+  }
+</script>
```
Comment (bm424) [inline src/opt-in.ts:79]: **praise:** keeping it simple - I like this.

**Review by bm424 — APPROVED** [review]

**Pull request merged**